src/util/text-utils.cc

```diff
@@ -243,12 +264,38 @@ bool ConvertStringToReal(const std::string &str,
     return true;
 #endif  // _MSC_VER
 
-  float f = KALDI_STRTOF(this_str, &end);
-  if (end != this_str)
-    while (isspace(*end)) end++;
-  if (end == this_str || *end != '\0' || errno != 0)
-    return false;
-  *out = f;
+  std::istringstream i(str);
+
+  if (!(i >> *out)) {
+    // Number conversion failed.
+    // I know it's not a numeric value.
+    // These lines are here just do check if str is nan or inf
+    const char *this_str = str.c_str();
+    char *end = NULL;
+    errno = 0;
+
+    float d = KALDI_STRTOF(this_str, &end);
```

Comment:
I wrote the code and the tests following your suggestion. 
I also completely removed strtof and strtod from the source code.